ubuntu20.04/nvidia-driver

```diff
@@ -47,15 +47,16 @@ _update_ca_certificates() {
 _resolve_kernel_version() {
     local version=$(apt-cache show "linux-headers-${KERNEL_VERSION}" 2> /dev/null | \
       sed -nE 's/^Version:\s+(([0-9]+\.){2}[0-9]+)[-.]([0-9]+).*/\1-\3/p' | head -1)
-    local flavor=$(echo ${KERNEL_VERSION} | sed 's/[^a-z]*//' | grep -Ev "^generic|virtual")
+    local kernel_flavor=$(echo ${KERNEL_VERSION} | sed 's/[^a-z]*//')
+    kernel_flavor="${kernel_flavor//virtual/generic}"
```

Comment:
So if I understand correctly, before this change we were changing any `generic-*` or `virtual` flavors to just `generic` (this caused issues with the `generic-64k` kernels since the `-64k` suffix was stripped. With your changes, we are only changing `virtual` flavors to `generic` -- which through your tests seems to be needed.